src/unleash-proxy.ts

```diff
@@ -44,11 +57,83 @@ export default class UnleashProxy {
         this.middleware = router;
 
         // Routes
-        router.get('/health', this.health.bind(this));
-        router.get('/', this.getEnabledToggles.bind(this));
-        router.post('/', this.lookupToggles.bind(this));
-        router.post('/client/metrics', this.registerMetrics.bind(this));
-        router.get('/client/features', this.unleashApi.bind(this));
+        router.get(
+            '/health',
+            openApiService.validPath({
+                responses: withStandardResponses(200, 503)(),
+            }),
+            this.health.bind(this),
+        );
+
+        router.get(
+            '/',
+            openApiService.validPath({
+                parameters: [
+                    ['appName', "Your application's name"],
+                    ['userId', "The current user's ID"],
+                    ['sessionId', "The current session's ID"],
+                    ['remoteAddress', "Your application's IP address"],
+                    ['properties', 'Additional properties'],
+                ].map(([k, v]) => ({
+                    name: k,
+                    description: v,
+                    in: 'query',
+                })),
+                responses: withStandardResponses(
+                    401,
+                    503,
+                )({ 200: featuresResponse }),
+            }),
+            this.getEnabledToggles.bind(this),
+        );
+
+        router.post(
+            '/',
+            openApiService.validPath({
+                requestBody: {
+                    content: {
+                        'application/json': {
+                            schema: {
+                                $ref: '#/components/schemas/featurePayloadSchema',
```

Comment:
Could put the `requestBody` in a file to keep the controller small, though may be a bit much.

* https://github.com/Unleash/unleash/blob/main/src/lib/routes/admin-api/project/features.ts#L139
* https://github.com/Unleash/unleash/blob/main/src/lib/openapi/spec/create-feature-request.ts

Reply:
Yeah, I'm happy to do this. Makes sense.